staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/cbor_test.go

```diff
@@ -0,0 +1,788 @@
+		},
+		{
+			in:       100000.0,
+			expected: "fa47c35000",
+		},
+		{
+			in:       3.4028234663852886e+38,
+			expected: "fa7f7fffff",
+		},
+		{
+			in:       1.0e+300,
+			expected: "fb7e37e43c8800759c",
+		},
+		{
+			in:       5.960464477539063e-8,
+			expected: "f90001",
+		},
+		{
+			in:       0.00006103515625,
+			expected: "f90400",
+		},
+		{
+			in:       -4.0,
+			expected: "f9c400",
+		},
+		{
+			in:       -4.1,
+			expected: "fbc010666666666666",
+		},
+		// TODO: Should Inf/-Inf/NaN be supported? Current Protobuf will encode this, but
+		// JSON will produce an error.  This is less than ideal -- we can't transcode
+		// everything to JSON.
+		{
+			in:       math.Inf(1),
+			expected: "f97c00",
+		},
+		{
+			in:       math.NaN(),
+			expected: "f97e00",
+		},
+		{
+			in:       math.Inf(-1),
+			expected: "f9fc00",
+		},
+		{
+			in:       false,
+			expected: "f4",
+		},
+		{
+			in:       true,
+			expected: "f5",
+		},
+		{
+			in:       nil,
+			expected: "f6",
+		},
+		{
+			in:       []interface{}{},
+			expected: "80",
+		},
+		{
+			in:       []interface{}{1, 2, 3},
+			expected: "83010203",
+		},
+		{
+			in:       []interface{}{1, []interface{}{2, 3}, []interface{}{4, 5}},
+			expected: "8301820203820405",
+		},
+		{
+			in:       []interface{}{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25},
+			expected: "98190102030405060708090a0b0c0d0e0f101112131415161718181819",
+		},
+		{
+			in:       map[string]interface{}{},
+			expected: "a0",
+		},
+	} {
+		t.Run(fmt.Sprintf("%T(%v)", tc.in, tc.in), func(t *testing.T) {
+			s := NewSerializer(nil, nil)
+			var buf bytes.Buffer
+			err := s.Encode(anyObject{tc.in}, &buf)
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			expected, err := hex.DecodeString("d9d9f7" + tc.expected)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if !bytes.Equal(expected, buf.Bytes()) {
+				t.Errorf("expected: %x\nactual: %x", expected, buf.Bytes())
+			}
+		})
+	}
+}
+
+func TestRecognizesData(t *testing.T) {
+	for _, tc := range []struct {
+		in string
+		ok bool
+	}{
+		{
+			in: "",
+			ok: false,
+		},
+		{
+			in: "d9",
+			ok: false,
+		},
+		{
+			in: "d9d9",
+			ok: false,
+		},
+		{
+			in: "d9d9f7",
+			ok: true,
+		},
+		{
+			in: "ffffff",
+			ok: false,
+		},
+		{
+			in: "d9d9f7000102030405060708090a0b0c0d0e0f",
+			ok: true,
+		},
+		{
+			in: "ffffff000102030405060708090a0b0c0d0e0f",
+			ok: false,
+		},
+	} {
+		t.Run(tc.in, func(t *testing.T) {
+			in, err := hex.DecodeString(tc.in)
+			if err != nil {
+				t.Fatal(err)
+			}
+			s := NewSerializer(nil, nil)
+			actual, unknown, err := s.RecognizesData(in)
+			if actual != tc.ok {
+				t.Errorf("expected recognized to be %t, got %t", tc.ok, actual)
+			}
+			if unknown {
+				t.Error("expected unknown to be false, got true")
+			}
+			if err != nil {
+				t.Errorf("expected nil error, got: %v", err)
+			}
+		})
+	}
+}
+
+func TestDecode(t *testing.T) {
```

Comment:
Explain what level of coverage you're trying to achieve here.  I see many examples, but I don't think I see nested maps or List types for instance.